### ISSUE TITLE
Manhunt

### ISSUE THREAD

**Issue by AlpyneDreams** [AlpyneDreams/d8vk#49]
"there might be a vram leak when certain camera effects are on screen? Unsure if it's d8vk" - Blisto

**Comment by Blisto91** [comment]
Finally got around to making a apitrace of this and it seems to show in the trace.

tldr of the issue: 
In the game there sometimes are short cutscenes where there will be a noise effect at the bottom and top of the screen and a blurry bar moving from top to bottom. 
If one observes the dxvk hud during these cutscenes one will notice the "Vidmen heap 0" used number just steadily keeps climbing. It will do this each time such a cutscene shows even if you've seen it before.

<details>
  <summary>Screenshot of the very first cutscene with the effect</summary>

![Screenshot_20221206_131311](https://user-images.githubusercontent.com/47954800/205909689-e4f0b39e-32a3-4f6a-9ac7-94e38dd6e492.png)
</details>

https://mega.nz/file/85hx3Jya#URV6iGGs1TPk5wypg8e3Qi2hnBnH9fa7yJRijC8m-zo

**Comment by WinterSnowfall** [comment]
Is this an apitrace that was captured on Windows? 

Indeed there are some unexpected object releases on replay, both with WineD3D and with d8vk + my refcount refactoring PR. Probably due to some off refcounts, the main branch doesn't exhibit these, but the memory usage spikes are still present. If anything, I believe the game itself is being stupid, maybe expecting us to free the objects that it owns.

Top memory usage is: 218 MB on main, 201MB with my PR (not a surprise, since object lifecycle is now a bit saner).

Does the memory usage go down once you get in game by any chance? I'm thinking maybe it simply delay-releases some objects, who knows.

**Comment by Blisto91** [comment]
Nein. Linux with d8vk if i remember. Will get a Windows native one

> Does the memory usage go down once you get in game by any chance? I'm thinking maybe it simply delay-releases some objects, who knows.

Not that i noticed. But will test some more.

**Comment by Blisto91** [comment]
Two fresh traces made on Windows. One with native and one with d8vk
Main menu -> play past first two camera with static scenes -> main menu -> same as before -> main menu and quit.

As can hopefully be seen the difference between native and d8vk (besides vram usage being MUCH higher with d8vk) is that with native vram will only go a bit up during gameplay and not really because of the static camera scenes, as expected, and will fall a bit again once you go back to the main menu. Choosing new game again won't change much from first run and vram will more or less stay the same.
With d8vk instead is that each time you view one of the camera scenes with static the vram will steadily go up and once you go back to main menu none of that will be released and number will be the same. Then when you start a new game again and view the same scenes vram will continue to go up even more.

https://mega.nz/file/oo5wBZQA#5D28PGL-NU9x6kLmq6MYIaMsTsFXwI1kaHPj9N6FKsU

**Comment by WinterSnowfall** [comment]
> As can hopefully be seen the difference between native and d8vk (besides vram usage being MUCH higher with d8vk)

Both native and d8vk traces look nearly identical, so it's probably a "behind the scenes problem". That one might perhaps debug with renderdoc, I'd expect, to figure out what exactly is bumping up VRAM usage in d8vk and why.

**Comment by Blisto91** [comment]
This is fixed in 1.0
